src/lib.rs

```diff
@@ -245,18 +243,13 @@ impl FileDialog {
   /// Shows the dialog if it is open. It is also responsible for state management.
   /// Should be called every ui update.
   pub fn show(&mut self, ctx: &Context) -> &Self {
-    if ctx.input().key_pressed(Key::Escape) {
-      self.state = State::Cancelled;
-    }
```

Comment:
This is actually important but I can see that the correct state was not being returned. Please put this back and just add a `return self`.

Reply:
Actually, it's probably better to put it at the start of the `State::Open => {` block below.

Reply:
I'll just fix it up after it's merged.